File: android/ValleTPV/app/src/main/java/com/valleapp/valletpv/tools/ServerConfig.java
package com.valleapp.valletpv.tools;

import android.content.ContentValues;

import org.json.JSONObject;

public class ServerConfig {
    String url =null ;
    String code = null;
    String UID = null;
    public ServerConfig(String code, String UID, String url){
        this.code = code;
        this.UID = UID;
        this.url = url;
    }
    public ServerConfig(String url){
        this.url = url;
    }
    public static ServerConfig loadJSON(String json){
        try {
            JSONObject obj = new JSONObject(json);
            return new ServerConfig(obj.getString("code"), obj.getString("UID"), obj.getString("url"));

        } catch (Exception e) {
            e.printStackTrace();
            return  null;
        }
    }

    public ContentValues getParams(){
        ContentValues params = new ContentValues();
        if (code != null && UID != null) {
            params.put("code", code);
            params.put("UID", UID);
        }
        return params;
    }

    public String getUrl(String endPoint){
        if (url == null || url.isEmpty()) return null;
        String strUrl = "";
        if(!url.contains("http://") && !url.contains("https://")) strUrl = "http://"+ url;
        if (!endPoint.startsWith("/")) endPoint = "/"+endPoint;
        if (strUrl.endsWith("/")) strUrl = strUrl.substring(0, strUrl.length()-1);
        if (!strUrl.endsWith("api")) strUrl = strUrl+"/api";
        return strUrl+endPoint;
    }
    public String toJson(){
        if (url != null && code != null && UID != null)
        return "{\"code\":\""+code+"\",\"UID\":\""+UID+"\",\"url\":\""+url+"\"}";
        else return null;
    }
}

File: android/ValleTPV/ValleTPVLib/src/main/java/com/valleapp/valletpvlib/tools/ServerConfig.kt
package com.valleapp.valletpvlib.tools

import org.json.JSONObject

data class ServerConfig(
    var codigo: String? = null,
    var uid: String? = null,
    var url: String? = null
) {

    fun getParseUrl(): String {
        var aux = parseUrl(url ?: "")
        if (!aux.endsWith("/")) {
            aux += "/"
        }
        if (!aux.endsWith("/api/")) {
            aux += "api/"
        }
        return aux
    }

    fun getUrlBase(): String {
        return parseUrl(url ?: "")
    }

    fun loadJSON(obj: Map<String, Any>) {
        try {
            codigo = obj["codigo"].toString()
            uid = obj["UID"].toString()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun isEmpty(): Boolean {
        return codigo.isNullOrEmpty() || uid.isNullOrEmpty() || url.isNullOrEmpty()
    }

    fun getParams(args: Map<String, Any>? = null): Map<String, String> {
        val aux = HashMap<String, String>()
        if (args != null && !isEmpty()) {
            codigo?.let { aux.put("codigo", it) }
            uid?.let { aux.put("UID", it) }
            for ((k, v) in args) {
                aux[k] = v.toString()
            }
        }
        return aux
    }

    fun toJson(): JSONObject? {
        return if (url != null && codigo != null && uid != null)
            JSONObject("{\"codigo\":\"$codigo\",\"UID\":\"$uid\",\"url\":\"$url\"}")
        else null
    }

    fun getWSUrl(): String {
        var newUrl = url?: ""
        newUrl = if (newUrl.contains("/api")) newUrl.substring(0, newUrl.indexOf("/api"))
        else "$newUrl/ws/"
        newUrl = when {
            newUrl.startsWith("http://") -> newUrl.replace("http://", "ws://")
            newUrl.startsWith("https://") -> newUrl.replace("https://", "wss://")
            else -> "ws://$newUrl"
        }
        return newUrl
    }


    companion object {
        fun parseUrl(url: String): String {
            var aux = url
            if (!aux.startsWith("https://") && !aux.startsWith("http://")) {
                aux = "http://$aux"
            }
            return aux
        }
    }
}
